src/Native/System.IO.Native/nativeio.cpp

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include "nativeio.h"
+#include <sys/stat.h>
+
+#if HAVE_STAT64 && !(defined(__APPLE__) && defined(_AMD64_))
+#    define stat_ stat64
+#    define fstat_ fstat64
+#else
+#   define stat_ stat
+#   define fstat_ fstat
+#endif
+
+static void ConvertFileStats(const struct stat_& src, FileStats* dst)
+{
+    dst->Flags = FILESTATS_FLAGS_NONE;
+    dst->Mode = src.st_mode;
```

Comment:
@weshaggard I'm not closing #2137 until these TODOs are done. I expect a lot of churn as this comes up and want to keep it moving forward incrementally. Separate issues for everything is going to be a bit painful.  

I've just added a [comment](https://github.com/dotnet/corefx/issues/2137#issuecomment-123800902) to #2137 where I'll gather TODOs as they come up and make sure to drive the TODOs to zero before closing out the issue. Does that approach work for you?